exotations/solvers/exotica_ik_solver/src/ik_solver.cpp

```diff
@@ -128,11 +111,24 @@ void IKSolver::Solve(Eigen::MatrixXd& solution)
             break;
         }
 
-        Eigen::MatrixXd Jinv = PseudoInverse(prob_->Cost.S * prob_->Cost.J);
-        Eigen::VectorXd qd = Jinv * yd;
-        if (UseNullspace)
-            qd += (Eigen::MatrixXd::Identity(prob_->N, prob_->N) - Jinv * prob_->Cost.S * prob_->J) *
-                  (q - prob_->qNominal);
+        if (C > 0)
+        {
+            yd.head(prob_->JN) = prob_->Cost.S * prob_->Cost.ydiff;
+            yd.tail(prob_->N) = q - qNominal;
+            J.topRows(prob_->JN) = prob_->Cost.S * prob_->Cost.J * W_ * (1.0 - C);
+            J.bottomRows(prob_->N) = C_;
+        }
+        else
+        {
+            yd = prob_->Cost.S * prob_->Cost.ydiff;
+            J = prob_->Cost.S * prob_->Cost.J * W_;
+        }
+
+#if EIGEN_VERSION_AT_LEAST(3, 3, 0)
+        qd = J.completeOrthogonalDecomposition().solve(yd);
+#else
+        qd = J.householderQr().solve(yd);
```

Comment:
Why not the `colPivHouseHolderQr` as used in Levenberg-Marquardt?

Reply:
Sure, for consistency but `fullPivHouseHolderQr` and `householderQr` should be equally good for most problems.